nemo/collections/common/data/lhotse/cutset.py

```diff
@@ -330,7 +332,17 @@ def parse_and_combine_datasets(
         if (w := item.get("weight")) is not None:
             weights.append(w)
 
-    assert all(t == tarred_status[0] for t in tarred_status), "Mixing tarred and non-tarred datasets is not supported."
+    all_same_tarred_status = all(t == tarred_status[0] for t in tarred_status)
```

Comment:
@pzelasko I propagated the two attributes and use them to decide whether to raise an error when mixing tarred and non-tarred datasets, do you think it's okay?